app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -211,14 +220,14 @@ def handle_telephony_result(method:, default:)
       end
     end
 
-    def track_events(otp_delivery_preference:)
+    def track_events(otp_delivery_preference:, otp_delivery_selection_result:)
       analytics.telephony_otp_sent(
         area_code: parsed_phone.area_code,
         country_code: parsed_phone.country,
         phone_fingerprint: Pii::Fingerprinter.fingerprint(parsed_phone.e164),
         context: context,
         otp_delivery_preference: otp_delivery_preference,
-        resend: params.dig(:otp_delivery_selection_form, :resend),
+        resend: otp_delivery_selection_result.extra[:resend],
```

Comment:
To avoid passing through the arguments, would it have worked to pick the resend property from the memoized form? Not sure I love the side-effect of property setting by `#submit`.

```suggestion
        resend: otp_delivery_selection_form.resend,
```

Reply:
We could have, yeah. I tried it initially, but it felt kinda weird to use the form rather than the result which is part of how the implementation ended up where it is.